pkg/controllers/scheduler/scheduler.go

```diff
@@ -339,16 +341,41 @@ func (s *Scheduler) reconcile(qualifiedName common.QualifiedName) (status worker
 			)
 			return worker.StatusError
 		}
-		profile, err := s.profileForFedObject(fedObject, s.buildFrameworkHandle())
+
+		var schedulingProfile *fedcorev1a1.SchedulingProfile
+		profileName := policy.GetSpec().SchedulingProfile
+		if len(profileName) > 0 {
+			schedulingProfile, err = s.schedulingProfileLister.Get(profileName)
+			if err != nil {
+				// TODO(hawjia): handle error - should we retry if not found?
+				keyedLogger.Error(err, "Failed to get scheduling profile")
+				s.eventRecorder.Eventf(
+					fedObject,
+					corev1.EventTypeWarning,
+					EventReasonScheduleFederatedObject,
+					"failed to schedule object: %v",
+					fmt.Errorf("failed to get scheduling profile: %w", err),
+				)
+
+				if apierrors.IsNotFound(err) {
+					return worker.StatusErrorNoRetry
```

Comment:
From the controller's perspective, this is not an error - the user specified a non-existent profile and we didn't find it. The controller did everything it can do. Shouldn't we return `StatusAllOK` here?

Reply:
Agreed.